### PR TITLE
Remove stacky, gappy penalties

### PR DESCRIPTION
**Pull request by yhahn**
TL;DR I think it's time to remove the gappy/stacky penalties.

In `master` we apply a very slight penalty and a very slight bonus in these scenarios:

- If you search for something using text that gives it additional context, we give a slight bonus to that feature ("stacking")
  - Example Query: "Atlanta Georgia"
    - "Atlanta Georgia" (the place, region) gets a slight text relev bonus in comparison to
    - "Atlanta Georgia's Best Beer Brewery" (the poi)
- If you search for something where the context is closer in hierarchy to the feature involved, it gets a higher bonus than a feature where the context is very far away in hierarchy
  - Example Query: "100 Main Street United States" << `address <=> country` context distance is far
  - Example Query: "100 Main Street Buffalo" << `address <=> place` context distance is short

These bonuses and penalties were meant to address a variety of corner case issues. One that I remember distinctly is this:

`12345 Georgia`

This could be a postcode search (the postcode `12345` in the state of `Georgia`) or maybe it's the beginning of an address (`12345 Georgia Ave`). In `master` we've decided to favor the former.

---

### A

For the query `69th Ave Oakland` we are currently matching

1. `69th Ave` (address) in `Oakland Gardens` (neighborhood) in `New York` (place)
2. `69th Ave` (address) in `Oakland` (place)

**Explanation:** 1 is beating 2 here because of the `gappy` penalty -- neighborhood and address are closer in our hierarchy than place and address. :disappointed:

**My personal response:** this feels super arbitrary and wrong.

### B

Roughly the same story. For the query `1509 16th St NW` we are currently matching

1. `1509 16th St` (address) in `Northwest Everett` (neighborhood) in `Everett` (place)
2. `1509 16th St NW` (address) in `Washington DC` (place)

**Explanation:** 1 is beating 2 here because of the `stacky` bonus. 2 is a contiguous query, while 1 has "additional context" ("NW" => "Northwest Everett") supplied to the address component.

**My personal response:** this also feels super arbitrary and wrong.

---

## Time to remove

Based on these cases and some IRL testing, the actual cases that stacky and gappy bonus/penalties were addressing seem to have been addressed through other means (e.g. @mattficke's work penalizing directional flips). For example, our stacky and gappy unit tests still pass with this functionality removed :smirk:

cc @mapbox/geocoding-gang

### CODE REVIEW

File: lib/verifymatch.js [hunk 1]
```diff
@@ -244,8 +244,6 @@ function verifyContexts(contexts, sets, indexes) {
```
Comment (yhahn) [inline lib/verifymatch.js]: This is the main substance of the PR, removing the stacky & gappy logic in verifymatch. The rest is all test update fallout (e.g. relevance scores that were 0.99 => 1.00).

Note that `geocode-unit.stacky.test.js` and `geocode-unit.gappy.test.js` still exist and pass.

**Comment by aarthykc** [comment]
This looks good to me @yhahn ! 🌮 , thank you for the explanation!
